deepspeed/runtime/pipe/engine.py

```diff
@@ -854,7 +855,7 @@ def _exec_send_activations(self, buffer_id):
         # NCCL does not like to send torch.BoolTensor types, so cast the mask to half().
         # We could do char, but with half() we can eventually flatten with other fp16
         # messages (TODO)
-        if self.module.__class__.__name__ == 'GPT2ModelPipe':
+        if self.module.__class__.__name__ == 'GPT2ModelPipe' or self.has_bool_tensors:
```

Comment:
Do we need to keep testing against GPT2ModelPipe here? Wont self.has_bool_tensors also be true? Or do we need to modify the GPT2ModelPipe codes?

Reply:
I considered backward compatibility.

Reply:
> do we need to modify the GPT2ModelPipe codes?

Yes I want to use the pipeline engine using another class. Supporting only one class name is inflexible.

Reply:
The reason I kept `GPT2ModelPipe` as is is for backwards compatibility.

Reply:
That makes sense. Thanks.